projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -88,6 +89,16 @@ function ( $track ) {
 				'tracks' => $tracks,
 			);
 
+			if ( $episode_options ) {
+				$player_data['options'] = array();
+				foreach ( $rss->get_items() as $episode ) {
+					$player_data['options'][] = array(
+						'label' => $this->get_plain_text( $episode->get_title() ),
+						'value' => $episode->get_id(),
+					);
+				}
+			}
```

Comment:
It looks like this duplicates what's already in the API response for tracks, but formats it server side.

Did you consider mapping tracks to options on the front-end, instead?

Reply:
The tracks are either the most recent 10, or the information about the selected track(s). This gets the basic title and GUID information for all available tracks. Originally I was sending back all the track information, but it was too much, and even formatting the JSON was proving too slow.

Reply:
Of course, I forgot about the 10 item limit on the track data. Thanks for clarifying.